src/extension-api-utils/extensionHost.ts

```diff
@@ -0,0 +1,54 @@
+import * as vscode from "vscode";
+
+declare const globalThis: {
+    [key: string]: any;
+};
+
+export interface HostWebviewPanel extends vscode.Disposable {
+    readonly webview: vscode.Webview;
+    readonly visible: boolean;
+    reveal(viewColumn?: vscode.ViewColumn, preserveFocus?: boolean): void;
+    readonly onDidChangeViewState: vscode.Event<any>;
+    readonly onDidDispose: vscode.Event<void>;
+}
+
+type LanguageRuntimeMetadata = Partial<{
+    // https://github.com/posit-dev/positron/blob/39a01b71/src/positron-dts/positron.d.ts#L357
+    /** The path to the runtime. */
+    runtimePath: string;
+
+    /**
+     * The fully qualified name of the runtime displayed to the user; e.g. "R 4.2 (64-bit)".
+     * Should be unique across languages.
+     */
+    runtimeName: string;
+
+    /**
+     * A language specific runtime name displayed to the user; e.g. "4.2 (64-bit)".
+     * Should be unique within a single language.
+     */
+    runtimeShortName: string;
+
+    /** The version of the runtime itself (e.g. kernel or extension version) as a string; e.g. "0.1" */
+    runtimeVersion: string;
+}>;
+
+export function getExtensionHostPreview(): void | ((url: string) => HostWebviewPanel) {
+    const pst = getPositronAPI();
+    if (!pst) { return; }
+    return (url: string) => pst.window.previewUrl(vscode.Uri.parse(url));
+}
+
+export async function getPositronPreferredRuntime(languageId: string): Promise<LanguageRuntimeMetadata | undefined> {
+    const pst = getPositronAPI();
+    if (!pst) { return; }
+    return await pst.runtime.getPreferredRuntime(languageId);
+}
+
+function getPositronAPI(): void | any {
+    if (!(typeof globalThis === "object")) { return; }
+    if (!("acquirePositronApi" in globalThis)) { return; }
+    if (typeof globalThis.acquirePositronApi !== "function") { return; }
+
+    return globalThis.acquirePositronApi();
```

Comment:
Just out of curiosity, not asking for a change... Is this all equivalent to `globalThis.acquirePositronApi?()`? (Other than you're gracefully dealing with `globalThis` not being an object and `acquirePositronApi` not being a function, both of which seem unlikely?)

Reply:
You mean `globalThis?.acquirePositronApi()`, right? Then yes, it's equivalent if overly cautious. I'm happy to switch to the cleaner syntax.

Reply:
Hmmm, I don't think that's right... I think `globalThis?.` checks for `globalThis` being null/undefined but not `acquirePositronApi`. It looks to me like `?()` is not a thing, I think maybe the original is best.

Or
```typescript
const apa = globalThis?.acquirePositronApi;
return apa ? apa() : undefined;
```

Or
```typescript
if (typeof acquirePositronApi === "function") {
  return acquirePositronApi();
}
```

(Bikeshedding so hard right now... sorry... :grimacing:)

Reply:
Or

```typescript
globalThis?.acquirePositronApi?.call(globalThis)
```
(Alright I'm done, I promise!)

Reply:
> I don't think that's right... I think `globalThis?.` checks for `globalThis` being null/undefined but not `acquirePositronApi`

Oh yeah, that's right. I think I'd rather revert back to something closer to what I had originally:

```js
function getPositronAPI(): undefined | any {
    if (typeof globalThis?.acquirePositronApi !== "function") { return; }

    return globalThis.acquirePositronApi();
}
```